web/source/kmwosk.ts

```diff
@@ -3622,7 +3622,12 @@ if(!window['keyman']['initialized']) {
       if('font' in layout) osk.fontFamily=layout['font']; else osk.fontFamily='';
 
       // Set flag to add default (US English) key label if specified by keyboard
-      layout.keyLabels=activeKeyboard && ((typeof(activeKeyboard['KDU']) != 'undefined') && activeKeyboard['KDU']);
+      if(typeof layout['displayUnderlying'] != 'undefined') {
+        layout.keyLabels = layout['displayUnderlying'] == true; // force bool
+      } else {
+        layout.keyLabels = activeKeyboard && ((typeof(activeKeyboard['KDU']) != 'undefined') && activeKeyboard['KDU']);
```

Comment:
So, what is the intended difference between `'displayUnderlying'` and `'KDU'`?  As best as I can tell, the affected lines are the only places where `'KDU'` is referenced.

Reply:
`KDU` is for the desktop .kvk file; `displayUnderlying` is the corresponding .keyman-touch-layout field, separated per platform.

Reply:
Correct. And `this.KDU` will then be used only for the desktop visual keyboard for new keyboards, and `displayUnderlying` will be used for touch keyboards. I allowed variance per platform since we may well want to show the underlying key on a tablet but hide on a phone for space and clarity reasons.

I could have used `KDU` for the `KVKL` section as well but trying to move away from short obscure variable names for new content.